ExitControllerTest.java

```diff
@@ -0,0 +1,160 @@
+package bcccp.carpark.exit;
+
+import bcccp.carpark.CarSensor;
+import bcccp.carpark.Carpark;
+import bcccp.carpark.IGate;
+import bcccp.tickets.adhoc.IAdhocTicket;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import org.junit.Before;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runners.MethodSorters;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+/**
+ *
+ * @author James Nocevski
+ */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class ExitControllerTest {
+    
+            Carpark mockCarpark;
+            IGate mockGate;
+            CarSensor mockIS;
+            CarSensor mockOS;
+            IExitUI mockUI;
+            IAdhocTicket mockTicket;
+            ExitController mockExit;
+    
+    
+            
+    @Before
+	public void setUp() {
+            mockCarpark = mock(Carpark.class);
+            mockGate = mock(IGate.class);
+            mockIS = mock(CarSensor.class);
+            mockOS = mock(CarSensor.class);
+            mockUI = mock(IExitUI.class);
+            mockTicket = mock(IAdhocTicket.class);
+            mockExit = new ExitController(mockCarpark, mockGate, mockIS, mockOS, mockUI);
+            
+	}
+    
+    @Test    
+    public void initialisationTest() {
+        assertTrue(mockExit instanceof ExitController);
+        assertTrue(mockIS instanceof CarSensor);
+        assertTrue(mockOS instanceof CarSensor);
+        assertEquals(mockExit.getState(), "IDLE");
+    }
+    
+    @Test
+    public void stateTransitionTestForAdhoc() {        //IDLE to WAITING, WAITING to REJECTED, WAITING to PROCESSED, PROCESSED to TAKEN, TAKEN to EXITING, EXITING to EXITED

```

Comment:
You need to break this method up.
You should only be testing one thing at a time so have something like IDLEtoWAITINGtest() and WAITINGtoREJECTEDtest()

Reply:
Fixed.